File: Quasics2019Code/AdaLovelace/src/main/cpp/Commands/TankDrive.cpp
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2017-2018 FIRST. All Rights Reserved.                        */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

#include "Commands/TankDrive.h"

TankDrive::TankDrive() {
  // Use Requires() here to declare subsystem dependencies
  // eg. Requires(Robot::chassis.get());
  Requires(Robot::driveBase.get());
}

// Called just before this Command runs the first time
void TankDrive::Initialize() {}

// Called repeatedly when this Command is scheduled to run
void TankDrive::Execute() {
  const double leftSpeed = Robot::oi->GetLeftDriveStick();
  const double rightSpeed = Robot::oi->GetRightDriveStick();
  double speedMult = 1;

  Robot::driveBase->SetMotorSpeed(leftSpeed * speedMult, rightSpeed * speedMult);
}

// Make this return true when this Command no longer needs to run execute()
bool TankDrive::IsFinished() { return false; }

// Called once after isFinished returns true
void TankDrive::End() {
  Robot::driveBase->Stop();
}

// Called when another command which requires one or more of the same
// subsystems is scheduled to run
void TankDrive::Interrupted() {
  End();
}

File: Experimental/2024/XRP/FromAlpha1/ReferenceProject/src/main/java/frc/robot/commands/TankDrive.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package frc.robot.commands;

import java.util.function.Supplier;

import edu.wpi.first.wpilibj2.command.Command;
import frc.robot.subsystems.Drivetrain;
import java.util.function.Supplier;

public class TankDrive extends Command {
  private final Drivetrain m_drivetrain;
  private final Supplier<Double> m_leftSpeedSupplier;
  private final Supplier<Double> m_rightRotateSupplier;

  /** Creates a new TankDrive. */
  public TankDrive(
    Drivetrain drivetrain,
    Supplier<Double> leftSpeedSupplier,
    Supplier<Double> rightRotateSupplier) {
  m_drivetrain = drivetrain;
  m_leftSpeedSupplier = leftSpeedSupplier;
  m_rightRotateSupplier = rightRotateSupplier;
  addRequirements(drivetrain);
  }

  // Called when the command is initially scheduled.
  @Override
  public void initialize() {}

  // Called every time the scheduler runs while the command is scheduled.
  @Override
  public void execute() {
    m_drivetrain.tankDrive(m_leftSpeedSupplier.get(), m_leftSpeedSupplier.get());
  }

  // Called once the command ends or is interrupted.
  @Override
  public void end(boolean interrupted) {}

  // Returns true when the command should end.
  @Override
  public boolean isFinished() {
    return false;
  }
}
